### PR TITLE
enh: use the appropriate CURL options for specifying methods

### PR DESCRIPTION
**Pull request by antoine-lizee**
cf #356.
A small fix can go a long way :-). I'd advocate for quick publishing to CRAN after inclusion of this PR (or whatever other code change that solves #356).

### CODE REVIEW

File: R/request.R [hunk 1]
```diff
@@ -99,7 +99,14 @@ print.request <- function(x, ...) {
 
 request_prepare <- function(req) {
   req <- request_combine(request_default(), req)
-  req$options$customrequest <- req$method
+  switch(req$method,
+         "POST" = {
```
Comment (hadley) [inline R/request.R]: Could you please make it more like `"POST" = (req$options$post <- TRUE)` ?

**Comment by hadley** [comment]
Would you mind also adding a bullet point to NEWS?

And any thoughts on how to test this?

File: R/request.R (hunk 1 above)
Comment (antoine-lizee) [inline R/request.R]: Why the brackets instead of curly? I'd rather have it be silent?

**Comment by antoine-lizee** [comment]
NEWS -> Will do
test -> Well, the problem came up with a wrong redirect behavior (see #356), and as you pointed out quite some time ago (cf https://github.com/Runscope/httpbin/issues/156), this is un-testable with httpbin (POSTing to `/redirect` throws a `405 METHOD NOT ALLOWED`).

File: R/request.R (hunk 1 above)
Comment (hadley) [inline R/request.R]: I'm not sure what you mean by silent, but I don't think choice of parens matters.

**Comment by hadley** [comment]
Since testing is hard, could you add a comment mentioning why this behaviour is important? I'm worried I'll forget about it and re-simplify in the future.

File: R/request.R (hunk 1 above)
Comment (antoine-lizee) [inline R/request.R]: Curly brackets are expressions, they return the result invisibly. Just try:
```
switch("a", "a" = (tt <- 2))
# [1] 2
switch("a", "a" = {tt <- 2})
# Nothing
```

Comment (hadley) [inline R/request.R]: Yes, but it really doesn't matter here. Regardless, splitting the expression over multiple lines really bugs me, so please fix.

**Comment by hadley** [comment]
Thanks!

**Comment by hadley** [comment]
Unfortunately this doesn't seem to be correct - setting `PUT` (which has been deprecated since curl 7.12.0) causes PUT requests to hang for forever on windows. Setting `UPLOAD`, which is the recommend option, cause it to hang forever on my computer too.

I think this is probably because I'm setting the post body parameters in `body_config()`.

**Comment by hadley** [comment]
A bit of googling suggests that using `CUSTOMREQUEST` is the correct way to do PUTs in libcurl

**Comment by antoine-lizee** [comment]
Good catch - this is officially a Windows exception to the cURL behaviour, itself being a standard but non-compliant way of dealing with RFC... Why is software getting so crazy? 
Thanks!

**Pull request merged**